### ISSUE TITLE
How do you compute loss in later two phase?

### ISSUE THREAD

**Issue by Johnson-yue** [youyuge34/PI-REC#14]
Hi, I read your paper , but I don't understand how to compute loss in later two phase? I know the G1_1 , G1_2, G1_3 is the same network but different phase.

In first phase:
                 X_fake1 = G1_1(E, M, Xgt)
                and computing loss.....

but in three phase:
              X_fake3 = G1_3(E,  X_fake2)
             and computing loss......         How I got X_fake2?

I guess 
method 1:
            X_fake_2 = G1_2(E, Cgt)
            X_fake3 = G1_3(E,  X_fake2)
            and computing loss...... 

method 2:
            torch.no_grad():
                    X_fake_2 = G1_2(E, Cgt)
            X_fake3 = G1_3(E,  X_fake2)
            and computing loss......

**Comment by youyuge34** [comment]
@Johnson-yue The three phases are not end-to-end, u can only start the next phase when the phase before is converged. So u can just save the model of well trained generator G1_2 to treat it as a fixed generator, and X_fake2 is generated from it and feed it to train a new G1_2 into G1_3.

**Comment by Johnson-yue** [comment]
ok， I see
